framework/tests/viewpointtest/AnnoOnTypeVariableUse.java

```diff
@@ -0,0 +1,17 @@
+import viewpointtest.quals.*;
+
+@SuppressWarnings("cast.unsafe.constructor.invocation")
+class AnnoOnTypeVariableUse<E> {
+    @ReceiverDependentQual E element;
+
+    static void test() {
+        AnnoOnTypeVariableUse<@B Element> d = new @A AnnoOnTypeVariableUse<@B Element>();
+        // d.element = @A |> @RDQ = @A
+        // thus expects no error here
+        d.element = new @A Element();
```

Comment:
Did you verify that this is failing as expected before this change?

Reply:
Can you add a test case with an non-RDQ annotation on a type variable use? E.g. add another field that is `@A E f` and make sure it still correctly gets replaced.

Some tests with expected errors also would increase confidence in this change.

Reply:
> Did you verify that this is failing as expected before this change?

Yes.

>Can you add a test case with an non-RDQ annotation on a type variable use? E.g. add another field that is @A E f and make sure it still correctly gets replaced.

Some tests with expected errors also would increase confidence in this change.

Sure, please see new commits.